### PR TITLE
Drive

### PR DESCRIPTION
**Pull request by PillageDev**
# Changed drive from percent drive to velocity drive

## Files changed...
- ArcadeDrive.java: Inverted y input to allow for correct controls
- RCFDrive.java: Inverted y input to allow for correct controls
- Rotate.java: Changed to velocity drive
- Drivetrain.java: Reset PID to 0, corrected Can Ids

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 1]
```diff
@@ -59,26 +59,26 @@ public class Drivetrain extends SubsystemBase {
     public static final double MAX_ENCODER_VELOCITY = 20743.0d;
     public static final double KF_TYPICAL_PERCENT_USAGE = 0.75d; // We will typically use 75% of max speed
     public static final double TALON_FULL_OUTPUT_SETTING = 1023;
-    public static final double VELOCITY_KF = 0.046d;
-    public static final double VELOCITY_KP = 0.03d;
+    public static final double VELOCITY_KF = 0.0d;
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:62]: We tested the PID today and found these numbers are very close. Let's leave them as is for now.

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 2]
```diff
@@ -273,10 +273,10 @@ public void periodic() {
          */
     }
 
-    public void percentDrive(double leftPercentage, double rightPercentage) {
-        leftTalonLead.set(ControlMode.PercentOutput, leftPercentage);
-        rightTalonLead.set(ControlMode.PercentOutput, rightPercentage);
-    }
+    //public void percentDrive(double leftPercentage, double rightPercentage) {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:276]: If this is not used, remove it

**Review by bjmcternan — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 1 above)
Comment (PillageDev) [reply src/main/java/frc/robot/subsystems/Drivetrain.java]: Fixed

File: src/main/java/frc/robot/subsystems/Drivetrain.java (hunk 2 above)
Comment (PillageDev) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:276]: Fixed

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -273,10 +273,10 @@ public void periodic() {
          */
```
[the remark this answers is not in the document]
Comment (bjmcternan) [reply src/main/java/frc/robot/subsystems/Drivetrain.java:273]: No, make this another ticket. You guys keep adding scope creep

**Review by rohanmishr — DISMISSED** [review]
Looks good

File: src/main/java/frc/robot/commands/Rotate.java [hunk 4]
```diff
@@ -54,7 +54,7 @@ public void execute() {
 
         m_lastError = m_error;
 
-        m_drivetrain.percentDrive(-percentOutput, percentOutput);
+        m_drivetrain.velocityDrive(-percentOutput, percentOutput);
```
Comment (cartha353) [inline src/main/java/frc/robot/commands/Rotate.java:57]: rotate relies on percent drive being used, also percent drive was deleted

Comment (bjmcternan) [reply src/main/java/frc/robot/commands/Rotate.java:57]: Yup, revert this change and add percent drive back

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 5]
```diff
@@ -234,11 +241,6 @@ public void periodic() {
...
     }
 
-    public void percentDrive(double leftPercentage, double rightPercentage) {
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:237]: Rotate.java needs percentDrive

**Review by bjmcternan — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/Rotate.java (hunk 4 above)
Comment (GoldCrafter) [reply src/main/java/frc/robot/commands/Rotate.java:57]: Added percent drive back in drivetrain and changed to percentdrive in rotate.java

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 6]
```diff
@@ -240,6 +240,11 @@ public void periodic() {
          * rightTalonLead.getClosedLoopError(0));
          */
     }
+    public void percentDrive(double leftPercentage, double rightPercentage) {
+
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:244]: Extra newline

File: src/main/java/frc/robot/subsystems/Drivetrain.java [hunk 7]
```diff
@@ -240,6 +240,11 @@ public void periodic() {
          * rightTalonLead.getClosedLoopError(0));
          */
     }
+    public void percentDrive(double leftPercentage, double rightPercentage) {
+        
+        leftTalonLead.set(ControlMode.PercentOutput, leftPercentage);
+        rightTalonLead.set(ControlMode.PercentOutput, rightPercentage);
+        }
```
Comment (bjmcternan) [inline src/main/java/frc/robot/subsystems/Drivetrain.java:247]: Fix tab spacing

**Review by bjmcternan — CHANGES_REQUESTED** [review]

**Review by bjmcternan — APPROVED** [review]

**Review by rohanmishr — APPROVED** [review]

**Pull request merged**